// cloudflare/resources/ai/to-markdown.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ToMarkdownTransformResponsesSinglePage = exports.ToMarkdown = void 0;
const resource_1 = require("../../resource.js");
const pagination_1 = require("../../pagination.js");
class ToMarkdown extends resource_1.APIResource {
    /**
     * Convert Files into Markdown
     */
    transform(file, params, options) {
        const { account_id } = params;
        return this._client.getAPIList(`/accounts/${account_id}/ai/tomarkdown`, ToMarkdownTransformResponsesSinglePage, {
            body: file,
            method: 'post',
            ...options,
            headers: { 'Content-Type': 'application/octet-stream', ...options?.headers },
            __binaryRequest: true,
        });
    }
}
exports.ToMarkdown = ToMarkdown;
class ToMarkdownTransformResponsesSinglePage extends pagination_1.SinglePage {
}
exports.ToMarkdownTransformResponsesSinglePage = ToMarkdownTransformResponsesSinglePage;
ToMarkdown.ToMarkdownTransformResponsesSinglePage = ToMarkdownTransformResponsesSinglePage;
