// cloudflare/resources/radar/ct/ct.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Ct = void 0;
const resource_1 = require("../../../resource.js");
const core_1 = require("../../../core.js");
const AuthoritiesAPI = __importStar(require("./authorities.js"));
const authorities_1 = require("./authorities.js");
const LogsAPI = __importStar(require("./logs.js"));
const logs_1 = require("./logs.js");
class Ct extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.authorities = new AuthoritiesAPI.Authorities(this._client);
        this.logs = new LogsAPI.Logs(this._client);
    }
    summary(dimension, query = {}, options) {
        if ((0, core_1.isRequestOptions)(query)) {
            return this.summary(dimension, {}, query);
        }
        return this._client.get(`/radar/ct/summary/${dimension}`, { query, ...options })._thenUnwrap((obj) => obj.result);
    }
    timeseries(query = {}, options) {
        if ((0, core_1.isRequestOptions)(query)) {
            return this.timeseries({}, query);
        }
        return this._client.get('/radar/ct/timeseries', { query, ...options })._thenUnwrap((obj) => obj.result);
    }
    timeseriesGroups(dimension, query = {}, options) {
        if ((0, core_1.isRequestOptions)(query)) {
            return this.timeseriesGroups(dimension, {}, query);
        }
        return this._client.get(`/radar/ct/timeseries_groups/${dimension}`, { query, ...options })._thenUnwrap((obj) => obj.result);
    }
}
exports.Ct = Ct;
Ct.Authorities = authorities_1.Authorities;
Ct.Logs = logs_1.Logs;
