// cloudflare/resources/zero-trust/dlp/entries/integration.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.IntegrationListResponsesSinglePage = exports.Integration = void 0;
const resource_1 = require("../../../../resource.js");
const pagination_1 = require("../../../../pagination.js");
class Integration extends resource_1.APIResource {
    /**
     * Integration entries can't be created, this will update an existing integration
     * entry This is needed for our generated terraform API
     *
     * @example
     * ```ts
     * const integration =
     *   await client.zeroTrust.dlp.entries.integration.create({
     *     account_id: 'account_id',
     *     enabled: true,
     *     entry_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/dlp/entries/integration`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Updates a DLP entry.
     *
     * @example
     * ```ts
     * const integration =
     *   await client.zeroTrust.dlp.entries.integration.update(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: 'account_id', enabled: true },
     *   );
     * ```
     */
    update(entryId, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/dlp/entries/integration/${entryId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Lists all DLP entries in an account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const integrationListResponse of client.zeroTrust.dlp.entries.integration.list(
     *   { account_id: 'account_id' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id } = params;
        return this._client.getAPIList(`/accounts/${account_id}/dlp/entries`, IntegrationListResponsesSinglePage, options);
    }
    /**
     * This is a no-op as integration entires can't be deleted but is needed for our
     * generated terraform API
     *
     * @example
     * ```ts
     * const integration =
     *   await client.zeroTrust.dlp.entries.integration.delete(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: 'account_id' },
     *   );
     * ```
     */
    delete(entryId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/dlp/entries/integration/${entryId}`, options)._thenUnwrap((obj) => obj.result);
    }
    /**
     * Fetches a DLP entry by ID.
     *
     * @example
     * ```ts
     * const integration =
     *   await client.zeroTrust.dlp.entries.integration.get(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: 'account_id' },
     *   );
     * ```
     */
    get(entryId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/dlp/entries/${entryId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.Integration = Integration;
class IntegrationListResponsesSinglePage extends pagination_1.SinglePage {
}
exports.IntegrationListResponsesSinglePage = IntegrationListResponsesSinglePage;
Integration.IntegrationListResponsesSinglePage = IntegrationListResponsesSinglePage;
