// cloudflare/resources/workers/beta/workers/versions.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.VersionsV4PagePaginationArray = exports.Versions = void 0;
const resource_1 = require("../../../../resource.js");
const pagination_1 = require("../../../../pagination.js");
class Versions extends resource_1.APIResource {
    /**
     * Create a new version.
     *
     * @example
     * ```ts
     * const version =
     *   await client.workers.beta.workers.versions.create(
     *     'worker_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    create(workerId, params, options) {
        const { account_id, deploy, ...body } = params;
        return this._client.post(`/accounts/${account_id}/workers/workers/${workerId}/versions`, {
            query: { deploy },
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List all versions for a Worker.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const version of client.workers.beta.workers.versions.list(
     *   'worker_id',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(workerId, params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/workers/workers/${workerId}/versions`, VersionsV4PagePaginationArray, { query, ...options });
    }
    /**
     * Delete a version.
     *
     * @example
     * ```ts
     * const version =
     *   await client.workers.beta.workers.versions.delete(
     *     'worker_id',
     *     'version_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    delete(workerId, versionId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/workers/workers/${workerId}/versions/${versionId}`, options);
    }
    /**
     * Get details about a specific version.
     *
     * @example
     * ```ts
     * const version =
     *   await client.workers.beta.workers.versions.get(
     *     'worker_id',
     *     'version_id',
     *     { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     *   );
     * ```
     */
    get(workerId, versionId, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(`/accounts/${account_id}/workers/workers/${workerId}/versions/${versionId}`, {
            query,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
}
exports.Versions = Versions;
class VersionsV4PagePaginationArray extends pagination_1.V4PagePaginationArray {
}
exports.VersionsV4PagePaginationArray = VersionsV4PagePaginationArray;
Versions.VersionsV4PagePaginationArray = VersionsV4PagePaginationArray;
