// cloudflare/resources/workers/beta/workers/workers.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkersV4PagePaginationArray = exports.Workers = void 0;
const resource_1 = require("../../../../resource.js");
const VersionsAPI = __importStar(require("./versions.js"));
const versions_1 = require("./versions.js");
const pagination_1 = require("../../../../pagination.js");
class Workers extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.versions = new VersionsAPI.Versions(this._client);
    }
    /**
     * Create a new Worker.
     *
     * @example
     * ```ts
     * const worker = await client.workers.beta.workers.create({
     *   account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *   name: 'my-worker',
     * });
     * ```
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/workers/workers`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Perform a complete replacement of a Worker, where omitted properties are set to
     * their default values. This is the exact same as the Create Worker endpoint, but
     * operates on an existing Worker. To perform a partial update instead, use the
     * Edit Worker endpoint.
     *
     * @example
     * ```ts
     * const worker = await client.workers.beta.workers.update(
     *   'worker_id',
     *   {
     *     account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     name: 'my-worker',
     *   },
     * );
     * ```
     */
    update(workerId, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/workers/workers/${workerId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * List all Workers for an account.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const worker of client.workers.beta.workers.list(
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * )) {
     *   // ...
     * }
     * ```
     */
    list(params, options) {
        const { account_id, ...query } = params;
        return this._client.getAPIList(`/accounts/${account_id}/workers/workers`, WorkersV4PagePaginationArray, {
            query,
            ...options,
        });
    }
    /**
     * Delete a Worker and all its associated resources (versions, deployments, etc.).
     *
     * @example
     * ```ts
     * const worker = await client.workers.beta.workers.delete(
     *   'worker_id',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    delete(workerId, params, options) {
        const { account_id } = params;
        return this._client.delete(`/accounts/${account_id}/workers/workers/${workerId}`, options);
    }
    /**
     * Perform a partial update on a Worker, where omitted properties are left
     * unchanged from their current values.
     *
     * @example
     * ```ts
     * const worker = await client.workers.beta.workers.edit(
     *   'worker_id',
     *   {
     *     account_id: '023e105f4ecef8ad9ca31a8372d0c353',
     *     logpush: true,
     *     name: 'my-worker',
     *     observability: {},
     *     subdomain: {},
     *     tags: ['my-team', 'my-public-api'],
     *     tail_consumers: [{ name: 'my-tail-consumer' }],
     *   },
     * );
     * ```
     */
    edit(workerId, params, options) {
        const { account_id, ...body } = params;
        return this._client.patch(`/accounts/${account_id}/workers/workers/${workerId}`, {
            body,
            ...options,
            headers: { 'Content-Type': 'application/merge-patch+json', ...options?.headers },
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Get details about a specific Worker.
     *
     * @example
     * ```ts
     * const worker = await client.workers.beta.workers.get(
     *   'worker_id',
     *   { account_id: '023e105f4ecef8ad9ca31a8372d0c353' },
     * );
     * ```
     */
    get(workerId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/workers/workers/${workerId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.Workers = Workers;
class WorkersV4PagePaginationArray extends pagination_1.V4PagePaginationArray {
}
exports.WorkersV4PagePaginationArray = WorkersV4PagePaginationArray;
Workers.WorkersV4PagePaginationArray = WorkersV4PagePaginationArray;
Workers.Versions = versions_1.Versions;
Workers.VersionsV4PagePaginationArray = versions_1.VersionsV4PagePaginationArray;
