// cloudflare/resources/brand-protection/brand-protection.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../resource.mjs";
import * as LogoMatchesAPI from "./logo-matches.mjs";
import { LogoMatches, } from "./logo-matches.mjs";
import * as LogosAPI from "./logos.mjs";
import { Logos } from "./logos.mjs";
import * as MatchesAPI from "./matches.mjs";
import { Matches, } from "./matches.mjs";
import * as QueriesAPI from "./queries.mjs";
import { Queries } from "./queries.mjs";
import { SinglePage } from "../../pagination.mjs";
export class BrandProtection extends APIResource {
    constructor() {
        super(...arguments);
        this.queries = new QueriesAPI.Queries(this._client);
        this.matches = new MatchesAPI.Matches(this._client);
        this.logos = new LogosAPI.Logos(this._client);
        this.logoMatches = new LogoMatchesAPI.LogoMatches(this._client);
    }
    /**
     * Return new URL submissions
     */
    submit(params, options) {
        const { account_id } = params;
        return this._client.post(`/accounts/${account_id}/brand-protection/submit`, options);
    }
    /**
     * Return submitted URLs based on ID
     */
    urlInfo(params, options) {
        const { account_id } = params;
        return this._client.getAPIList(`/accounts/${account_id}/brand-protection/url-info`, BrandProtectionURLInfoResponsesSinglePage, options);
    }
}
export class BrandProtectionURLInfoResponsesSinglePage extends SinglePage {
}
BrandProtection.BrandProtectionURLInfoResponsesSinglePage = BrandProtectionURLInfoResponsesSinglePage;
BrandProtection.Queries = Queries;
BrandProtection.Matches = Matches;
BrandProtection.Logos = Logos;
BrandProtection.LogoMatches = LogoMatches;
